acceptance/suites/pre_suite/puppet4_compat/70_install_puppet.rb

```diff
@@ -1,5 +1,18 @@
 require 'puppetserver/acceptance/compat_utils'
 
+# zypper installs of puppet-agent for SLES will fail if the package can't be
+# validated with the Puppet GPG key.  Beaker doesn't implicitly install the
+# Puppet GPG key so need to install it before trying to install the
+# puppet-agent.
+step "Install Updated Puppet GPG key for Any SLES Agents."
+
+nonmaster_agents.each { |agent|
+  variant = master['platform'].to_array.first
```

Comment:
I just modified the initial commit for this since it was small enough, now 7a4cedc.